lib/jsdom/living/nodes/Node-impl.js

```diff
@@ -167,6 +167,8 @@ class NodeImpl extends EventTargetImpl {
       newChildImpl._ownerDocument = this._ownerDocument;
     }
 
+    // https://dom.spec.whatwg.org/#concept-node-ensure-pre-insertion-validity
```

Comment:
Not necessary in this pull request, but it would be really nice to factor out these things into helper functions, like they are in the spec, instead of inlining them into the methods of the Node class. This has been a long-standing problem.